enterprise/backend/src/metabase_enterprise/transforms/util.clj

```diff
@@ -55,6 +56,18 @@
   [transform]
   (= :python (-> transform :source :type keyword)))
 
+(defn transform-source-type
+  "Returns the type of a transform's source: :python, :native, or :mbql.
+  Throws if the source type cannot be detected."
+  [source]
+  (case (keyword (:type source))
+    :python :python
+    :query  (if (lib.query/native? (:query source))
```

Comment:
Switched this from `lib/native-only-query?` to `lib.query/native?` to match the `native-query-transform?` above. But not really sure which I should prefer here.

Reply:
I think both should be `lib/native-only-query?`, that's testing if this is a query that would have been `:native` in MBQL4.